xscen/io.py

```diff
@@ -493,6 +490,235 @@ def coerce_attrs(attrs):
+            if not isinstance(cols, pd.MultiIndex)
+            else [nm or varname_lvl for nm in cols.name]
+        )
+        if isinstance(df_data.columns, pd.MultiIndex) or isinstance(
+            cols, pd.MultiIndex
+        ):
+            # handle different depth of multicolumns, expand MultiCol of coord with None for missing levels.
+            cols = pd.MultiIndex.from_arrays(
+                [
+                    cols.get_level_values(lvl) if lvl in cols.names else [None]
+                    for lvl in df_data.columns.names
+                ],
+                names=df_data.columns.names,
+            )
+        dfc.columns = cols
+        dfs.append(
+            dfc[~dfc.index.duplicated()]
+        )  # We dropped columns thus the index is not unique anymore
+    dfs.append(df_data)
+    return pd.concat(dfs, axis=1).sort_index(level=row, key=season_sort_key)
+
+
+def to_table(
+    ds: Union[xr.Dataset, xr.DataArray],
+    *,
+    row: Union[None, str, Sequence[str]] = None,
+    column: Union[None, str, Sequence[str]] = None,
+    sheet: Union[None, str, Sequence[str]] = None,
+    coords: Union[bool, Sequence[str]] = True,
+) -> Union[pd.DataFrame, dict]:
+    """Convert a dataset to a pandas DataFrame with support for multicolumns and multisheet.
+
+    This function will trigger a computation of the dataset.
+
+    Parameters
+    ----------
+    ds : xr.Dataset or xr.DataArray
+      Dataset or DataArray to be saved.
+      If a Dataset with more than one variable is given, the dimension "variable"
+      must appear in one of `row`, `column` or `sheet`.
+    row : str or sequence of str, optional
+      Name of the dimension(s) to use as indexes (rows).
+      Default is all data dimensions.
+    column : str or sequence of str, optional
+      Name of the dimension(s) to use as columns.
+      Default is "variable", i.e. the name of the variable(s).
+    sheet : str or sequence of str, optional
+      Name of the dimension(s) to use as sheet names.
+    coords: bool or sequence of str
+      A list of auxiliary coordinates to add to the columns (as would variables).
+      If True, all (if any) are added.
+
+    Returns
+    -------
+    pd.DataFrame or dict
+      DataFrame with a MultiIndex with levels `row` and MultiColumn with levels `column`.
+      If `sheet` is given, the output is dictionary with keys for each unique "sheet" dimensions tuple, values are DataFrames.
+      The DataFrames are always sorted with level priority as given in `row` and in ascending order,.
+    """
+    if isinstance(ds, xr.Dataset):
+        da = ds.to_array(name="data")
+        if len(ds) == 1:
+            da = da.isel(variable=0).rename(data=da.variable.values[0])
+
+    def _ensure_list(seq):
+        if isinstance(seq, str):
+            return [seq]
+        return list(seq)
+
+    row = _ensure_list(row or ([d for d in da.dims if d != "variable"]))
+    column = _ensure_list(column or (["variable"] if len(ds) > 1 else []))
```

Comment:
I think there are issues here with automatically filling the row, colums and sheet. It looks like they are filled even if passed somewhere else.

One example:
I tried with a 2d dataset with dims `time` x `geom` and many variables.
`df= xs.io.to_table(ds1, sheet=['variable'])` fails with 

`ValueError: Repeated dimension names. Got row=['time', 'geom'], column=['variable'] and sheet=['variable'].Each dimension should appear only once.`

Reply:
I think my last commit makes better guesses for the dimensions!